jsonschema_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2017 johandorland ( https://github.com/johandorland )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gojsonschema
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+type jsonSchemaTest struct {
+	Description string `json:"description"`
+	// Some tests do not pass yet, so some tests are manually edited to include
+	// an extra attribute whether that specific test should be disabled and skipped
+	Disabled bool                 `json:"disabled"`
+	Schema   interface{}          `json:"schema"`
+	Tests    []jsonSchemaTestCase `json:"tests"`
+}
+type jsonSchemaTestCase struct {
+	Description string      `json:"description"`
+	Data        interface{} `json:"data"`
+	Valid       bool        `json:"valid"`
+}
+
+func TestSuite(t *testing.T) {
+
+	wd, err := os.Getwd()
+	if err != nil {
+		panic(err.Error())
+	}
+	wd = filepath.Join(wd, "testdata")
+
+	go func() {
+		err := http.ListenAndServe(":1234", http.FileServer(http.Dir(filepath.Join(wd, "remotes"))))
+		if err != nil {
+			panic(err.Error())
+		}
+	}()
+
+	testDirectories := []string{"draft4", "draft6", "draft7"}
```

Comment:
I wonder if it might not be a better idea to perform a directory glob in a well-defined base testDirecotory seeking for all directories named "draft\d+". That way this becomes more portable/extensible by the next developer to simply create a new directory with json test files. This current method is not a show-stopper, however. I'm just merely making a suggestion.

Reply:
You're right, since writing the tests I came across a few things that I could redo a bit nicer and this is one of them. Ideally I would just want to run all tests in `draft\d+` recursively, but unfortunately there are some tests in the official test suite which will probably never be supported by this library (like zero terminated floats not being valid integers). While testing for example I often manually added "draft7/optional" to see which things didn't work yet. Anyway long story short, this is not perfect yet.